tests/model_selection/test_incremental.py

```diff
@@ -188,22 +193,42 @@ def additional_calls(scores):
 
 
 @gen_cluster(client=True)
-def test_search(c, s, a, b):
+def test_search_basic(c, s, a, b):
     X, y = make_classification(n_samples=1000, n_features=5, chunks=(100, 5))
     model = SGDClassifier(tol=1e-3, loss="log", penalty="elasticnet")
 
     params = {"alpha": np.logspace(-2, 2, 100), "l1_ratio": np.linspace(0.01, 1, 200)}
 
-    search = IncrementalSearch(model, params, n_initial_parameters=10, max_iter=10)
+    search = IncrementalSearchCV(model, params, n_initial_parameters=20, max_iter=10)
     yield search.fit(X, y, classes=[0, 1])
 
-    assert search.history_results_
-    for d in search.history_results_:
+    assert search.history_
+    for d in search.history_:
         assert d["partial_fit_calls"] <= search.max_iter + 1
     assert isinstance(search.best_estimator_, SGDClassifier)
     assert search.best_score_ > 0
     assert "visualize" not in search.__dict__
     assert search.best_params_
+    assert search.cv_results_ and isinstance(search.cv_results_, dict)
+    assert {
+        "mean_partial_fit_time",
+        "mean_score_time",
+        "std_partial_fit_time",
+        "std_score_time",
+        "test_score",
+        "rank_test_score",
+        "model_id",
+        "params",
+        "partial_fit_calls",
+        "param_alpha",
+        "param_l1_ratio",
+    }.issubset(set(search.cv_results_.keys()))
+    assert all(isinstance(v, np.ndarray) for v in search.cv_results_.values())
+    assert (
```

Comment:
This is failing for me occasionally (locally too).

Reply:
The immediate failure is because

```
(Pdb) pp pd.DataFrame(search.cv_results_).iloc[[5, 12]][['model_id', 'test_score']]
    model_id  test_score
5          5        0.68
12        12        0.65
```

But looking at the history, I can't convince myself that we should have picked model 12 as the best.

Reply:
I can get this to fail to on my local machine. When I run the tests until failure then plot `search.history_` for a subset of model IDs, I get

<img width="325" alt="screen shot 2018-10-15 at 10 25 36 am" src="https://user-images.githubusercontent.com/1320475/46960780-c98a0800-d08e-11e8-8e90-3d6732b91e0b.png">

Model 10 had the highest final score, so the same assert statement fails, but model 16 is the model corresponding to `search.best_index_`. This happens because the scores decrease over time. I view this as a test of the adaptive algorithm, not a test of basic `IncrementalSearchCV` functionality.

I believe the search in this test should be passive, or use `decay_rate=0` because of the small dataset and simple model. The adaptive algorithm removed some models, and the score decreased over time (=> models removed had higher scores than retained models).

I think there are two methods to resolve it:

1. Set `decay_rate=0` (which would require parts of #406, 8c7153d929c5fc67355301f26e9be566944b9ec0 and b5ce47c1d1bb973a46a471aac14c30d4d997499b).
2. Implement the test change in https://github.com/dask/dask-ml/pull/370#issuecomment-428788259 then revert it in #406.

Reply:
I don't think a probabilistic test like implemented in 
https://github.com/dask/dask-ml/pull/370#issuecomment-428788259 is good.

I think on IncrementalSearchCV we can't really make any claims about the relative scores of the models. We can't claim that the "best" model (the model who was trained through to the last round) has the highest score (average or final). I think we just drop this assert from the test.

Reply:
> I don't think a probabilistic test like implemented in #370 (comment) is good.

I agree.

> I think on IncrementalSearchCV we can't really make any claims about the relative scores of the models

For passive incremental searches, we can make claims about the relative scores of models.
For adaptive incremental searches, we can not make claims about the relative scores of models.

We can do this because completely passive searches (`patience=False`) don't drop models. Every model is fitted for `max_iter` calls to `partial_fit`. All of these models are considered in `_get_best`, and the model with the highest final score is selected.

I'll merge 8c7153d929c5fc67355301f26e9be566944b9ec0 and b5ce47c1d1bb973a46a471aac14c30d4d997499b later today.

Reply:
Are  8c7153d and b5ce47c related to this specific issue?

We should have a test with the default params, and I think this is a good test, aside from the assertion that `search.cv_results_["test_score"][search.best_index_]` has at least the highest score.